addons/binding/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/job/SunPhaseJob.java

```diff
@@ -30,11 +30,13 @@
     @Override
     protected void executeJob(String thingUid, JobDataMap jobDataMap) {
         AstroThingHandler astroHandler = AstroHandlerFactory.getHandler(thingUid);
-        Channel phaseNameChannel = astroHandler.getThing().getChannel(CHANNEL_ID_SUN_PHASE_NAME);
-        if (astroHandler != null && phaseNameChannel != null) {
-            SunPhaseName phaseName = (SunPhaseName) jobDataMap.get(KEY_PHASE_NAME);
-            ((Sun) astroHandler.getPlanet()).getPhase().setName(phaseName);
-            astroHandler.publishChannelIfLinked(phaseNameChannel.getUID());
+        if (astroHandler != null) {
+            Channel phaseNameChannel = astroHandler.getThing().getChannel(CHANNEL_ID_SUN_PHASE_NAME);
+            if (phaseNameChannel != null) {
```

Comment:
astroHandler could never by null on the old line 34 because a NPE would have been thrown at the old line 33